app/src/main/java/ch/epfl/sdp/drone3d/service/impl/weather/OpenWeatherWeatherService.kt

```diff
@@ -0,0 +1,88 @@
+package ch.epfl.sdp.drone3d.service.impl.weather
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.epfl.sdp.drone3d.R
+import ch.epfl.sdp.drone3d.model.weather.WeatherReport
+import ch.epfl.sdp.drone3d.service.api.weather.WeatherService
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.qualifiers.ApplicationContext
+import org.json.JSONObject
+import timber.log.Timber
+import java.net.URL
+import java.util.*
+import javax.inject.Inject
+import kotlin.concurrent.thread
+
+/**
+ * WeatherService implementation using OpenWeather API
+ */
+class OpenWeatherWeatherService @Inject constructor(@ApplicationContext val context: Context) :
+    WeatherService {
+
+    companion object {
+        private const val API_ENDPOINT = "https://api.openweathermap.org/data/2.5/weather?"
+        private const val TEMP_UNIT = "metric"
+        private val NO_DATA = WeatherReport("N/A", "N/A", 0.0, 0, 0.0, 0, Date())
+    }
+
+    override fun getWeatherReport(location: LatLng): LiveData<WeatherReport> {
+        val liveData = MutableLiveData<WeatherReport>()
+        thread {
+            val data = fetchAPI(location)
+            val report = parseJsonToWeatherReport(data)
+            liveData.postValue(report)
+        }
+        return liveData
+    }
+
+    /**
+     * Parse a json string into a WeatherReport
+     */
+    private fun parseJsonToWeatherReport(jsonReport: String): WeatherReport {
+        var report = NO_DATA
+        try {
+            /* Extracting JSON returns from the API */
+            val jsonObj = JSONObject(jsonReport)
+            val main = jsonObj.getJSONObject("main")
+            val wind = jsonObj.getJSONObject("wind")
+            val weather = jsonObj.getJSONArray("weather").getJSONObject(0)
+
+            val updatedAt = Date(jsonObj.getLong("dt") * 1000)
```

Comment:
Why doing the *1000?

Reply:
The value given by the json is in miliseconds, and the *1000 is there to convert it in seconds